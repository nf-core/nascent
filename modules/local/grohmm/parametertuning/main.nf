process GROHMM_PARAMETERTUNING {
    tag "$meta.id"
    label 'process_high'
    label 'process_long'

    conda "conda-forge::r-base=4.1.1 conda-forge::r-optparse=1.7.1 conda-forge::r-argparse=2.1.3 bioconda::bioconductor-genomicfeatures=1.46.1 bioconda::bioconductor-grohmm=1.28.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-e9a6cb7894dd2753aff7d9446ea95c962cce8c46:0a46dae3241b1c4f02e46468f5d54eadcf64beca-0' :
        'quay.io/biocontainers/mulled-v2-e9a6cb7894dd2753aff7d9446ea95c962cce8c46:0a46dae3241b1c4f02e46468f5d54eadcf64beca-0' }"

    input:
    tuple val(meta), path(bams), path(bais)
    path gtf
    path tune_parameter_file

    output:
    path "*.tuning.csv" , emit: tuning
    path  "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    parameter_tuning.R \\
        --bam_file ${bam} \\
        --tuning_file ${tune_parameter_file} \\
        --outprefix ${prefix} \\
        --gtf $gtf \\
        --outdir ./ \\
        --cores $task.cpus \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-grohmm: \$(Rscript -e "library(groHMM); cat(as.character(packageVersion('groHMM')))")
    END_VERSIONS
    """
}
