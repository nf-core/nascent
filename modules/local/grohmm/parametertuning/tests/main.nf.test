nextflow_process {

    name "Test Process GROHMM_PARAMETERTUNING"
    script "modules/local/grohmm/parametertuning/main.nf"
    process "GROHMM_PARAMETERTUNING"

    test("Should fail with a tuning file") {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'Sall' ],
                    [ file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S0mR1.bam", checkIfExists: true),
                    file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S40mR1.bam", checkIfExists: true) ],
                    [],
                    [],
                ]
                input[1] = file(
                            "https://gist.github.com/edmundmiller/c142801995689ed8d15ebcf40b2fb042/raw/eca3b955312209b5845cca084bb506d5250b3d33/hg19.chr7.refGene.gtf",
                            checkIfExists: true
                            )
                """
            }
        }

        then {
            assert process.failed
        }
    }

    test("Should run with a tuning file and without bai files") {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'Sall' ],
                    [ file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S0mR1.bam", checkIfExists: true),
                    file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S40mR1.bam", checkIfExists: true) ],
                    [],
                    file("${projectDir}/tests/config/tuningparams_small.csv", checkIfExists: true),
                ]
                input[1] = file(
                            "https://gist.github.com/edmundmiller/c142801995689ed8d15ebcf40b2fb042/raw/eca3b955312209b5845cca084bb506d5250b3d33/hg19.chr7.refGene.gtf",
                            checkIfExists: true
                            )
                """
            }
        }

        then {
            // TODO
            assert process.failed
        }
    }
}
