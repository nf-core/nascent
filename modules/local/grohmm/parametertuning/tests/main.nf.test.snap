{
    "versions": {
        "content": [
            {
                "GROHMM_PARAMETERTUNING": {
                    "r-base": "4.3.3",
                    "bioconductor-grohmm": "1.39.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-24T11:44:17.664038682"
    },
    "Should run with defaults": {
        "content": [
            [
                "\"LtProbB\",\"UTS\",\"merged\",\"dissociated\",\"total\",\"errorRate\",\"txSize\"",
                "-100,5,10,0,10,0.00617665225447807,921"
            ],
            "Sall_5_-100.tuning.consensus.bed:md5,013b67a2aef7956b6400a61d413f65a8"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T20:21:05.448531"
    }
}